src/pages/auth/LoginPage.js

```diff
@@ -4,69 +4,76 @@
  * @author Ken Pham
  */
 
-import React from 'react';
-import {useDispatch, useSelector} from 'react-redux'
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
 
 // Firebase
-import firebase from 'firebase';
-import {firebase_auth} from '../../firebase-config';
+import firebase from "firebase";
+import { firebase_auth } from "../../firebase-config";
 
 // Styles
-import './styles/LoginPage.css'
-import FeedbackImage from '../../components/FeedbackImage';
+import "./styles/LoginPage.css";
+import FeedbackImage from "../../components/FeedbackImage";
 
 // Actions
-import * as authActions from '../../store/actions/auth/auth'
-import { FACEBOOK_LOGIN, GOOGLE_LOGIN } from '../../constants/auth';
+import * as authActions from "../../store/actions/auth/auth";
+import { FACEBOOK_LOGIN, GOOGLE_LOGIN } from "../../constants/auth";
 
 /**
  * @summary Rendering the page users use to login
- * @param {} props 
+ * @param {} props
  * @returns JSX Elements
  */
-const LoginPage = props => {
+const LoginPage = (props) => {
+  const userType =
+    props.location.state && props.location.state.userType
+      ? props.location.state.userType
+      : "NO TYPE";
 
-    const userType = (props.location.state && props.location.state.userType) ? props.location.state.userType : "NO TYPE"
+  const dispatch = useDispatch();
+  const userData = useSelector((state) => state.authReducer.userData);
 
-    const dispatch = useDispatch();
-    const userData = useSelector(state => state.authReducer.userData);
+  React.useEffect(() => {
+    if (userData.isAuth && userData.isAuth === true) {
+      props.history.push("/");
+      window.location.reload(false);
```

Comment:
Even though the data on Firebase got updated immediately after login, the data from the global store won't update (as it doesn't fetch again). So the page needs to reload so that the data got fetched again, and that data (on the top right corner) will be updated.